src/inmanta/agent/in_process_executor.py

```diff
@@ -52,18 +467,17 @@ async def create_executor(
         :param executor_id: executor identifier containing an agent name and a blueprint configuration.
         :return: An Executor instance
         """
+        if agent_name in self.executors:
+            out = self.executors[agent_name]
+        else:
+            async with self._creation_locks.get(agent_name):
+                if agent_name in self.executors:
+                    out = self.executors[agent_name]
+                else:
+                    out = InProcessExecutor(agent_name, agent_uri, self.environment, self.client, self.eventloop, self.logger)
+                    self.executors[agent_name] = out
+        assert out.uri == agent_uri
```

Comment:
This is not yet always true, we need cleanup with the agent instance. 
I think that should do